code/modules/atmospherics/FEA_fire.dm

```diff
@@ -1,79 +1,83 @@
 #define HOTSPOT_MEDIUM_LIGHTS
 
+/// Exposes our reagents and material to some temperature, letting them figure out how to react to it.
 /atom/proc/temperature_expose(datum/gas_mixture/air, exposed_temperature, exposed_volume)
-	if (reagents)
-		reagents.temperature_reagents(exposed_temperature, exposed_volume, 350, 300, 1)
+	if (src.reagents)
+		src.reagents.temperature_reagents(exposed_temperature, exposed_volume, 350, 300, 1)
 	if (src.material)
 		src.material.triggerTemp(src, exposed_temperature)
-	return null
 
-/turf/proc/hotspot_expose(exposed_temperature, exposed_volume, soh,electric = 0)
+/// We reacts to the exposed temperature, calls [/atom/proc/temperature_expose] on everything within us, and exposes fluids to electricity if need be.
+/turf/proc/hotspot_expose(exposed_temperature, exposed_volume, soh, electric = FALSE)
 	SHOULD_CALL_PARENT(TRUE)
 	if (src.material)
 		src.material.triggerTemp(src, exposed_temperature)
-	if (reagents)
-		reagents.temperature_reagents(exposed_temperature, exposed_volume, 350, 300, 1)
+	if (src.reagents)
+		src.reagents.temperature_reagents(exposed_temperature, exposed_volume, 350, 300, 1)
 	if(!ON_COOLDOWN(src, "hotspot_expose_to_atoms__1", 1 SECOND) || !ON_COOLDOWN(src, "hotspot_expose_to_atoms__2", 1 SECOND) || !ON_COOLDOWN(src, "hotspot_expose_to_atoms__3", 1 SECOND) || !ON_COOLDOWN(src, "hotspot_expose_to_atoms__4", 1 SECOND) || !ON_COOLDOWN(src, "hotspot_expose_to_atoms__5", 1 SECOND))
 		if (electric) //mbc : i'm putting electric zaps on here because eleczaps ALWAYS happen alongside hotspot expose and i dont want to loop all atoms twice
-			for (var/atom/item in src) //I hate having to add this here too but too many things use hotspot_expose. This might cause lag on large fires.
+			for (var/atom/movable/item as anything in src)
 				item.temperature_expose(null, exposed_temperature, exposed_volume)
 				if (item?.flags & FLUID_SUBMERGE)
 					item.electric_expose(electric)
 		else
-			for(var/atom/item in src) //I hate having to add this here too but too many things use hotspot_expose. This might cause lag on large fires.
+			for(var/atom/movable/item as anything in src)
 				item.temperature_expose(null, exposed_temperature, exposed_volume)
 
-
-
-/turf/simulated/hotspot_expose(exposed_temperature, exposed_volume, soh, electric = 0)
-	. = ..()
-	var/datum/gas_mixture/air_contents = return_air()
-
+/// * Checks if we should light on fire if we do not have a hotspot already. If we should and don't have one, spawns one.
+/// * Returns: TRUE if we ignited or already have a hotspot, FALSE if we didn't make one or have one.
+/turf/simulated/hotspot_expose(exposed_temperature, exposed_volume, soh, electric = FALSE)
```

Comment:
I've asked dev chat if anyone knows what `soh` means by the way, I have no clue lol

Reply:
i have a faint idea on what it does and even fainter idea on why it does it. i have no idea what to remain soh to.

Reply:
sounds like "standard of heating" or suchlike. h could be hotspot too. really poor naming lol

Reply:
someone said set own hotspot but im not sure how accurate that is

Reply:
think we got it down in imcoder, renaming and resolving